sw/host/tests/chip/spi_passthru/src/main.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::time::Duration;
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::spiflash::SpiFlash;
+use opentitanlib::test_utils::spi_passthru::{ConfigJedecId};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::uart::console::UartConsole;
+use opentitanlib::{execute_test};
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+
+    #[structopt(
+        long, parse(try_from_str=humantime::parse_duration),
+        default_value = "600s",
+        help = "Console receive timeout",
+    )]
+    timeout: Duration,
+
+    #[structopt(
+        long,
+        default_value = "BOOTSTRAP",
+        help = "Name of the debugger's SPI interface",
+    )]
+    spi: String,
+
+}
+
+fn test_jedec_id(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let config = ConfigJedecId {
+        device_id: 0x1234,
+        manufacturer_id: 0x56,
+        continuation_code: 0x7f,
+        continuation_len: 3,
+    };
+    config.execute(&*uart)?;
+
+    let spi = transport.spi(&opts.spi)?;
+    let jedec_id = SpiFlash::read_jedec_id(&*spi, 16)?;
+    log::info!("jedec_id = {:x?}", jedec_id);
+    assert_eq!(jedec_id, [
+        // 3 continuation codes of 0x7F.
+        0x7f, 0x7f, 0x7f,
+        // Manufacturer ID of 0x56.
+        0x56,
+        // Device ID 0x1234 (in little endian order).
+        0x34, 0x12,
+        // All zeros up to read length.
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
```

Comment:
At some point, we might want to loosen this particular restriction and only take the slice that is valid data. I'm not sure if SPI flash chips are required to return any particular value after reading beyond the ID bytes.

Reply:
Added a comment explaining that the zeros are an OT specific behavior.